src/pages/CommandView/commandViewHelpers.tsx

```diff
@@ -0,0 +1,244 @@
+): JSX.Element | undefined => {
+  if (!namespace) {
+    return <Navigator message={'systems'} link={'/systems'} />
+  }
+  if (!systemName) {
+    return (
+      <Navigator
+        message={`systems : ${namespace}`}
+        link={`/systems/${namespace}`}
+      />
+    )
+  }
+  if (!version || !commandName || !system || !command) {
+    return (
+      <Navigator
+        message={`systems : ${namespace} : ${systemName}`}
+        link={`/systems/${namespace}/${systemName}`}
+      />
+    )
+  }
+
+  return undefined
+}
+
+/**
+ * Determine if a Parameter is a Bytes parameter or, in the case of
+ * Dictionaries, if any sub-parameter is a Bytes parameter
+ *
+ * @param parameter - the Parameter to check
+ * @returns true or false
+ */
+const isByte = (parameter: Parameter): boolean => {
+  return (
+    parameter.type === 'Bytes' ||
+    (parameter.parameters.length > 0
+      ? isByteCommand(parameter.parameters)
+      : false)
+  )
+}
+
+/**
+ * Determine if any of the parameters of a command are bytes parameters
+ *
+ * @param parameters - the parameter array to check
+ * @returns true or false
+ */
+const isByteCommand = (parameters: Parameter[]) => {
+  return parameters.map(isByte).some((x) => x)
+}
+
+/**
+ * Handle resetting the model to initial values by retaining only
+ * bytes parameters values.
+ *
+ * @param initialData - the empty version of the model
+ * @param model - the updated version of the model
+ * @param parameters - the parameters of the command
+ * @returns an empty model that retains the values for bytes parameters
+ */
+const handleByteParametersReset = (
+  initialData: ObjectWithStringKeys,
+  model: ObjectWithStringKeys,
+  parameters: Parameter[],
+) => {
+  if ('parameters' in model) {
+    const updatedInitialData = { ...initialData }
+    const { parameters: updatedParameters } = updatedInitialData as {
+      parameters: ObjectWithStringKeys
+    }
+    const modelParameters = model['parameters'] as
+      | ObjectWithStringKeys
+      | undefined
+
+    if (modelParameters) {
+      for (const parameter of parameters) {
+        if (!isByte(parameter)) continue
+
+        const { key } = parameter
+
+        if (key in modelParameters) {
+          updatedParameters[key] = modelParameters[key]
+        }
+      }
+
+      return {
+        ...initialData,
+        parameters: updatedParameters,
+      }
+    }
+  }
+
+  return initialData
+}
+
+/* Pull the filename out of a dataURL string */
+const cleanUrlData = (urlData: string) => {
+  const parts = urlData.split(';')
+
+  if (parts.length > 1) {
+    return parts[1].split('=')[1]
+  }
+
+  return urlData
+}
+
+/**
+ * Fix up the model for display in the UI so that Bytes type object don't show
+ * up as extremely long strings
+ *
+ * @param model - the model to clean up
+ * @param parameters - the parameters for the command
+ * @returns cleaned up model
+ */
+const cleanModelForDisplay = (
+  model: ObjectWithStringKeys,
+  parameters: Parameter[],
```

Comment:
I think the confusion *might* be coming from the difficulty in mentally parsing identical syntax that means one thing in one context and another thing in a different context. This is a shortcoming of typescript IMO. Maybe it would help if we bring this up on the big screen and walk through it together.